include/seqan3/core/platform.hpp

```diff
@@ -34,20 +44,29 @@
 //  C++ standard and features
 // ============================================================================
 
+#if SEQAN3_DOXYGEN_ONLY(1)0
+//!\brief This disables the warning you would get if you compile with `-std=c++17`.
+#define SEQAN3_DISABLE_CPP17_DIAGNOSTIC
+#endif // SEQAN3_DOXYGEN_ONLY(1)0
+
 // C++ standard [required]
 #ifdef __cplusplus
-    static_assert(__cplusplus >= 201703, "SeqAn3 requires C++17, make sure that you have set -std=c++17.");
+#   if (__cplusplus < 201703)
+#       error "SeqAn3 requires C++20, make sure that you have set -std=c++2a (gcc9) or -std=c++20 (gcc10 and higher)."
+#   elif not defined(SEQAN3_DISABLE_CPP17_DIAGNOSTIC) && (__cplusplus >= 201703) && (__cplusplus < 201709)
+#      pragma GCC warning "SeqAn 3.1.x is the last version that supports C++17. Newer SeqAn versions, including this one, might not compile with -std=c++17. To disable this warning, use -std=c++2a (gcc9), -std=c++20 (gcc10 and higher), or -DSEQAN3_DISABLE_CPP17_DIAGNOSTIC."
```

Comment:
why is this a `pragma` and not an `error`?

Reply:
Same as with pre-cxx-11 ABI.
It might compile, but no guarantees from our side.
It would be totally OK to use cpp17 **for now**, unless you use the `view::char_valid_for` or `view::assign_strictly_to` anywhere.